### PR TITLE
develop-core

### PR DESCRIPTION
**Pull request by venkatesh9959**

### CODE REVIEW

File: src/Components/Button.tsx [hunk 1]
```diff
@@ -6,20 +6,21 @@
...
- *  It used as Cancel  Button or Submit button or Reset button
- *
- */
+
 interface ButtonProps {
   type?: "submit" | "reset" | "button";
   value?: string;
   ButtonText?: string;
-  onClickHandler?: () => void;
+  onClickHandler?: any;
```
Comment (ims7inc) [inline src/Components/Button.tsx:14]: why this type is any?

File: src/Components/Button.tsx [hunk 2]
```diff
@@ -6,20 +6,21 @@
...
   addclass?: string;
 }
 
+/**
+ *  Button  Component Is Used  for Button Fields
+ *  It used as Cancel  Button or Submit button or Reset button
+ *
+ */
 const Button = (props: ButtonProps) => {
   const { type, value, ButtonText, onClickHandler, disable, addclass } = props;
```
Comment (ims7inc) [inline src/Components/Button.tsx:25]: ButtonText to buttonText

File: src/Pages/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
+import React, { useState } from "react";
+import InputText from "../Components/InputText"
+import Button from "../Components/Button";
+import "./Login.scss";
+import { Link,useNavigate} from "react-router-dom";
+import axios from "axios";
+
+const Login = () => {
+    const [open, setOpen] = React.useState(false);
```
Comment (ims7inc) [inline src/Pages/Login.tsx:9]: why did you used React.useState here? you can use useState

File: src/Pages/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        history("/dashboard");
+    };
+
+    const onSubmit = (e: any) => {
+        e.preventDefault();
+
+        var emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        const { Email, Password } = Login;
+        const LoginDetails = { Email, Password };
+        if (Email == undefined && Password == undefined) {
```
Comment (ims7inc) [inline src/Pages/Login.tsx:33]: check for undefined & "" as well **if (!Email && !Password)**

File: src/Pages/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
+import React, { useState } from "react";
+import InputText from "../Components/InputText"
+import Button from "../Components/Button";
+import "./Login.scss";
+import { Link,useNavigate} from "react-router-dom";
+import axios from "axios";
+
+const Login = () => {
+    const [open, setOpen] = React.useState(false);
+    const handleToggle = () => {
```
Comment (ims7inc) [inline src/Pages/Login.tsx:10]: toggle for what? give meaning full name

File: src/Pages/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        setOpen(!open);
+    };
+    const handleClose = () => {
```
Comment (ims7inc) [inline src/Pages/Login.tsx:13]: close for what?

File: src/Pages/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        setOpen(false);
+    };
+    const history = useNavigate ();
+    const [login, setLogin] = useState<any>({});
+    const onchageLogin = (key: string, value: string) => {
```
Comment (ims7inc) [inline src/Pages/Login.tsx:18]: onLoginChange

**Review by ims7inc — CHANGES_REQUESTED** [review]

File: src/Pages/Login/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+          localStorage.setItem("userId", response.data.Result.Id);
+          localStorage.setItem("firstName", response.data.Result.FirstName);
+          localStorage.setItem("lastName", response.data.Result.LastName);
+          // window.location.href = "/dashboard";
+          MovetoDashboard();
+        }
+      });
+
+      return true;
```
Comment (ims7inc) [inline src/Pages/Login/Login.tsx:54]: why return statement here?

File: src/Pages/Login/Login.tsx [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+    } else {
+      alert("invalid emil address");
+      return false;
```
Comment (ims7inc) [inline src/Pages/Login/Login.tsx:57]: why return?

File: src/Pages/Login/Login.tsx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+                          addclasses={"form-control-lg input-lg"}
+                          onChageHandler={(e: any) =>
+                            onLoginChange("Password", e.target.value)
+                          }
+                        />
+                        <div className="form-group row">
+                          <div className="col-md-6 col-12 text-center text-md-left">
+                            <fieldset>
+                              <input
+                                type="checkbox"
```
Comment (ims7inc) [inline src/Pages/Login/Login.tsx:104]: on remember me what is th functionality ?

**Review by ims7inc — CHANGES_REQUESTED** [review]

File: src/Pages/Login/Login.tsx (hunk 8 above)
Comment (venkatesh9959) [reply src/Pages/Login/Login.tsx:54]: Line :37 condtion false  ayithe return false ....
retutn rayaothe page next page veluthundi
Line:54 conditin is true ayithe return true

**Pull request closed without merging**